src/function/table.rs

```diff
@@ -226,10 +248,14 @@ impl Table {
         true
     }
 
-    /// Returns the entries at the given index if the entry is live and the index in bounds.
-    pub(crate) fn get_index(&self, i: usize) -> Option<(&[Value], &TupleOutput)> {
+    /// Returns the entries at the given index if the entry is live (and possibly not subsumed) and the index in bounds.
+    pub(crate) fn get_index(
+        &self,
+        i: usize,
+        include_subsumed: bool,
+    ) -> Option<(&[Value], &TupleOutput)> {
         let (inp, out) = self.vals.get(i)?;
-        if !inp.live() {
+        if !inp.live() || (!include_subsumed && out.subsumed) {
```

Comment:
I'm a little less sure about the changes to `table.rs`. Should `Table` internalize the knowledge of whether some tuple is subsumed, or we always handle them at GJ/Query level? If it is the former, does it suffice to just change this one place? Other methods like `iter` and `iter_range` do not take an `include_subsumed` flag. Could you maybe give more rationale on this change?

Reply:
Sometimes we want to filter by subsumption and sometimes not... In GJ for example it depends on whether we are running a check or matching on a rule.

I didn't include that check in the `iter` since most of the use cases for `iter` (and its relatives) do include subsumed I believe, such as making the tries and indices.

I think I will update it though to make that arg explicit to `iter` et al, to mirror the API.

Reply:
Thanks for this feedback! I added the flag to the other iter methods as well and I think it's cleaner now.